mokapot/model.py

```diff
@@ -564,16 +583,32 @@ def _get_starting_labels(psms, model):
 
     elif model.is_trained:
         try:
-            scores = model.estimator.decision_function(psms.features)
+            scores = model.decision_function(psms)
         except AttributeError:
             scores = model.estimator.predict_proba(psms.features).flatten()
 
+        # Get starting_labels and feat_pass from pre-trained model
         start_labels = psms._update_labels(scores, eval_fdr=model.train_fdr)
-        LOGGER.info(
-            "\t- The pretrained model found %i PSMs at q<=%g.",
-            (start_labels == 1).sum(),
-            model.train_fdr,
-        )
+        feat_pass = (start_labels == 1).sum()
+
+        # Get starting_labels and feat_pass from best feature
+        feat_res = psms._find_best_feature(model.train_fdr)
+        _, feat_pass_best_feat, start_labels_best_feat, _ = feat_res
+
+        if feat_pass > feat_pass_best_feat:
+            LOGGER.info(
+                "\t- The pretrained model found %i PSMs at q<=%g.",
+                feat_pass,
+                model.train_fdr,
+            )
+        else:
+            logging.warning(
+                "Learned model did not improve upon the pretrained "
+                "input model. Now re-scoring each collection of PSMs "
+                "using the best feature."
+            )
+            start_labels = start_labels_best_feat
+            feat_pass = feat_pass_best_feat
```

Comment:
This looks good, but it strikes me that maybe we should cache the results of `_find_best_feature` as part of the object. This is something I'll look into, so no action needed by you.